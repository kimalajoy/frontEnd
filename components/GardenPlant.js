import React from 'react';
import { View, Text, TouchableOpacity} from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import styles from '../styles/styles';

export default function GardenPlant ({ info, navigation }) {

  return (
    <TouchableOpacity
      style={styles.myGardenBody}
      onPress={() => {
        navigation.navigate('PlantProfilePage', {id: info.id});
        navigation.dangerouslyGetParent().setOptions({ tabBarVisible: false })}}
        testID="GardenPlant.TouchableOpacity">
        <View style={styles.iconContainer}>
          <MaterialCommunityIcons
            style={styles.icon}
            name="flower"
            size={50}
            testID="GardenPlant.Icon"/>
        </View>
        <View>
          <Text style={[styles.textLight, styles.gardenPlantText]}>{info.plant_name}</Text>
        </View>
    </TouchableOpacity>
  );
}
