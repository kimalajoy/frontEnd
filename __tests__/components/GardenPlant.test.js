import React from 'react';
import { render, cleanup, fireEvent } from 'react-native-testing-library';

import GardenPlant from '../../components/GardenPlant';

describe("Garden Plant", () => {
  let gardenPlant;
  const onPressMock = jest.fn();

  const info = {
    plant_name: 'Test Plant Name'
  }

  beforeEach(() => {

    gardenPlant = render(<GardenPlant
        info={info}
      />)
  });

  afterEach(cleanup);

  test('it renders touchable opacity', () => {
    const touchable = gardenPlant.getByTestId('GardenPlant.TouchableOpacity')

    expect(touchable).toBeTruthy()
  })

  test('it renders icon', () => {
    const icon = gardenPlant.getByTestId('GardenPlant.Icon')

    expect(icon).toBeTruthy()
  })

  test('it renders name of plant', () => {
    const name = gardenPlant.getByText('Test Plant Name')

    expect(name).toBeTruthy
  })

  test('it renders / snapshot test', () => {
    expect(gardenPlant).toMatchSnapshot()
  })
})
