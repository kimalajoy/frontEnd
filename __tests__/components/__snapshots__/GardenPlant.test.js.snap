// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Garden Plant it renders / snapshot test 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 180,
      "opacity": 1,
      "width": 110,
    }
  }
  testID="GardenPlant.TouchableOpacity"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 50,
          },
          Object {
            "color": "limegreen",
            "textShadowColor": "grey",
            "textShadowOffset": Object {
              "height": 12,
              "width": 1,
            },
            "textShadowRadius": 5,
          },
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="GardenPlant.Icon"
    >
      ï‰Š
    </Text>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#fafafa",
            "fontSize": 19,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      Test Plant Name
    </Text>
  </View>
</View>
`;
